Documentation/Common-Gotchas.md

```diff
@@ -0,0 +1,232 @@
+# Common Gotchas
+
+There are some common errors that arise when trying to use Argo. Many are due to
+the fact that Argo is so fundamentally different from how you might be used to
+decoding JSON, and others are due to limitations of Swift. This document aims
+to address some of those situations and set you on the right track!
+
+## `curry` limitations
+
+It is very important to fully curry your model’s `init` in order to use Argo.
+Because of this, `Argo` recommends using
+[`Curry`](https://github.com/thoughtbot/Curry) which allows you to change your
+standard `init` functions into fully curried functions. Unfortunately, Swift
+can only compile `curry` for up to 20 arguments in a reasonable amount of time.
```

Comment:
how about "for around 20 arguments"? I'd like to avoid needing to update this doc if we make incremental improvements to `curry`.